lib/matplotlib/backend_tools.py

```diff
@@ -318,7 +321,12 @@ class ToolQuit(ToolBase):
     default_keymap = rcParams['keymap.quit']
 
     def trigger(self, sender, event, data=None):
-        Gcf.destroy_fig(self.figure)
+        try:
+            manager = self.figure.canvas.manager
+        except:
```

Comment:
Doing the internal try/except is the right thing to do here.  Failing to find an attribute should raise an `AttributeError`, if we are implementing attribute look up by delegating to dict lookup, it is our job to convert the `KeyError` -> `AttributeError` before it escapes our code base.  

Changing exception types is an API break so should get this right up front.